_studio/mfx_lib/shared/src/mfx_brc_common.cpp

```diff
@@ -1617,7 +1641,26 @@ mfxStatus ExtBRC::GetFrameCtrl (mfxBRCFrameParam* par, mfxBRCFrameCtrl* ctrl)
             qpMin = std::min(qp - 1, qpMin);
     }
     ctrl->QpY = qp - m_par.quantOffset;
+    if (m_par.mMBBRC )
+    {
+        if (ctrl->NumExtParam == 0)
+        {
+            //attach MBBRC buffer
+            ctrl->NumExtParam = 1;
+            ctrl->ExtParam = &(m_ExtBuff[par->EncodedOrder % m_ExtBuff.size()]);
+        }
+        mfxExtMBQP * pExtMBQP = (mfxExtMBQP*)Hevc_GetExtBuffer(ctrl->ExtParam, ctrl->NumExtParam, MFX_EXTBUFF_MBQP);
+        if (pExtMBQP)
+        {
+            //fill QP map
+            for (size_t i = 0; i < pExtMBQP->NumQPAlloc; i++)
+            {
+                pExtMBQP->QP[i] = (mfxU8)(qp +((qp<51)? (i%2):0));
```

Comment:
could you clarify what is happening? why we need such adjustment? I think comment in code will be good

Reply:
It is example of using MBQP in ExtBRC. 
Daria, please add comment